projects/plugins/wpcomsh/woa.php

```diff
@@ -269,3 +269,97 @@ function wpcomsh_woa_post_process_maybe_enable_wordads( $args, $assoc_args ) {
 	WP_CLI::success( 'WordAds options transferred and module activated' );
 }
 add_action( 'wpcomsh_woa_post_transfer', 'wpcomsh_woa_post_process_maybe_enable_wordads', 10, 2 );
+
+/**
+ * Checks for WooCommerce connection details, validates them, and stores them in the database.
+ *
+ * @param array $args       Positional arguments.
+ * @param array $assoc_args Named arguments.
+ */
+function wpcomsh_woa_post_process_store_woocommerce_connection_details( $args, $assoc_args ) {
+	$woocommerce_connection_details = WP_CLI\Utils\get_flag_value( $assoc_args, 'store-woocommerce-connection-details', false );
+	if ( ! $woocommerce_connection_details ) {
+		return;
+	}
+
+	// Validate that we have a valid JSON object.
+	$woocommerce_connection_details_decoded = json_decode( $woocommerce_connection_details, true );
+	if ( ! is_array( $woocommerce_connection_details_decoded ) ) {
+		WP_CLI::warning( 'Invalid WooCommerce connection details provided: ' . $woocommerce_connection_details );
+		return;
+	}
+
+	$valid_keys = array(
+		'auth'           => array(
+			'access_token',
+			'access_token_secret',
+			'site_id',
+			'user_id',
+			'updated',
+		),
+		'auth_user_data' => array(
+			'email',
+		),
+	);
+
+	$required_root_keys = array( 'auth' );
+
+	foreach ( $required_root_keys as $required_root_key ) {
+		if ( ! isset( $woocommerce_connection_details_decoded[ $required_root_key ] ) ) {
+			WP_CLI::warning( 'Invalid WooCommerce connection details provided. Missing ' . $required_root_key );
+			return;
+		}
+	}
+
+	$unexpected_root_keys = array_diff( array_keys( $woocommerce_connection_details_decoded ), array_keys( $valid_keys ) );
+	if ( ! empty( $unexpected_root_keys ) ) {
+		WP_CLI::warning( 'Invalid WooCommerce connection details provided. Unexpected root key(s): ' . implode( ', ', $unexpected_root_keys ) );
+		return;
+	}
+
+	$option_data = array();
+
+	foreach ( $valid_keys as $valid_key => $required_key_fields ) {
+		if ( ! isset( $woocommerce_connection_details_decoded[ $valid_key ] ) ) {
+			// If the data isn't present, keep going - we validate presence for required keys above.
+			continue;
+		}
+
+		if ( ! is_array( $woocommerce_connection_details_decoded[ $valid_key ] ) ) {
+			WP_CLI::warning( 'Invalid WooCommerce connection details provided. Missing ' . $valid_key );
+			return;
+		}
+
+		if ( count( $required_key_fields ) !== count( $woocommerce_connection_details_decoded[ $valid_key ] ) ) {
```

Comment:
If the connection details API response is updated in the future, even with optional keys, this will fail. 

Maybe we should continue instead of `return`?

Reply:
I've updated this in e90a98cb20bc07aa80fe5e45650a4eadd5b885be to keep processing the fields -- we're only accepting the ones we know about, so this should be safe.